### PR TITLE
Feature/server/227 server define and implement user roles

### LINKED ISSUE #227: Server: Define and Implement User Roles
Define and implement roles and access rights in the server component including web frontend for:

- Scenario Admin
- Gamemaster
- Application Admin (Dev/Maintainer)

Restrict API endpoints to certain roles, where necessary.

This issue does not include a configuration interface.

### PR DESCRIPTION
**Pull request by simohlsen**
Closes #227

This PR implements the following roles for users of the web interface along with methods to enforce them:
- Web-Admin (Maintainer/Developer)
- Scenario-Admin
- Game-Master
- None (Read-only)

Note: This implementation assume a fixed hierarchy between the roles. So a role with more power than another can do all things the other can.

### CODE REVIEW

File: server/dbsetup.py [hunk 1]
```diff
@@ -220,7 +220,7 @@ def __takes_part_in():
 def __create_executions():
     insert(Execution(id=3, name="Übungssimulation \"Busunglück\" 2024",
                      scenario_id=0))
-    insert(Execution(id=23456, name="Kreativer Name", scenario_id=1))
+    #insert(Execution(id=23456, name="Kreativer Name", scenario_id=1))  # FIXME: broken scenario
```
Comment (YIllmann) [inline server/dbsetup.py:223]: delete. useless anyway

**Review by YIllmann — APPROVED** [review]

**Review by H1ghBre4k3r — APPROVED** [review]
LGTM

File: server/utils/decorator.py [hunk 2]
```diff
@@ -61,28 +63,67 @@ def wrapper(*args, **kwargs):
 
             kwargs[arg] = value
             return f(*args, **kwargs)
+
         return wrapper
+
     return decorator
 
 
 def admin_only(func: Callable):
```
Comment (7erra) [inline server/utils/decorator.py:72]: This function now seems like a repetition of `role_required`. IMO this should be removed.

File: server/scenario/web_api/scenario.py [hunk 3]
```diff
@@ -1,13 +1,11 @@
 from flask import Blueprint
 
 import models
-from utils.decorator import admin_only
 
 web_api = Blueprint("web_api-scenario", __name__)
 
 
 @web_api.get("/templates")
-@admin_only
```
Comment (7erra) [inline server/scenario/web_api/scenario.py:10]: Why not protect all endpoints even if they are "only" GET requests? Sensitive data might get leaked.

**Review by 7erra — CHANGES_REQUESTED** [review]

**Review by 7erra — APPROVED** [review]

**Pull request merged**